src/main/scala/io/iohk/ethereum/mining/Miner.scala

```diff
@@ -203,4 +216,11 @@ object Miner {
   val MaxNonce: BigInt = BigInt(2).pow(64)
 
   val DagFilePrefix: ByteString = ByteString(Array(0xfe, 0xca, 0xdd, 0xba, 0xad, 0xde, 0xe1, 0xfe).map(_.toByte))
+
+  sealed trait MineResult {
+    def triedHashes: Int
+  }
+  case class MineSuccessful(override val triedHashes: Int, pow: ProofOfWork, nonce: ByteString) extends MineResult
```

Comment:
We can omit `override val`.

Reply:
I personally prefer to always add these, but I can remove it if you insist

Reply:
I'm with Christos on this, because it's a case class